src/lib.rs

```diff
@@ -189,3 +219,238 @@ impl From<Options> for original::Options {
+            }
+            (Self::Image(l0), original::Tag::Image(r0_link_type, r0_cow_str_1, r0_cow_str_2)) => {
+                l0.0 == *r0_link_type
+                    && l0.1 == r0_cow_str_1.clone().into_string()
+                    && l0.2 == r0_cow_str_2.clone().into_string()
+            }
+            _ => false,
+        }
+    }
+}
+
+impl PartialEq<original::Alignment> for Alignment {
+    fn eq(&self, other: &original::Alignment) -> bool {
+        matches!(
+            (self, other),
+            (Alignment::None, original::Alignment::None)
+                | (Alignment::Left, original::Alignment::Left)
+                | (Alignment::Center, original::Alignment::Center)
+                | (Alignment::Right, original::Alignment::Right)
+        )
+    }
+}
+
+impl PartialEq<original::LinkType> for LinkType {
+    fn eq(&self, other: &original::LinkType) -> bool {
+        matches!(
+            (self, other),
+            (LinkType::Inline, original::LinkType::Inline)
+                | (LinkType::Reference, original::LinkType::Reference)
+                | (
+                    LinkType::ReferenceUnknown,
+                    original::LinkType::ReferenceUnknown
+                )
+                | (LinkType::Collapsed, original::LinkType::Collapsed)
+                | (
+                    LinkType::CollapsedUnknown,
+                    original::LinkType::CollapsedUnknown
+                )
+                | (LinkType::Shortcut, original::LinkType::Shortcut)
+                | (
+                    LinkType::ShortcutUnknown,
+                    original::LinkType::ShortcutUnknown
+                )
+                | (LinkType::Autolink, original::LinkType::Autolink)
+                | (LinkType::Email, original::LinkType::Email)
+        )
+    }
+}
+
+impl PartialEq<original::CodeBlockKind<'_>> for CodeBlockKind {
+    fn eq(&self, other: &original::CodeBlockKind<'_>) -> bool {
+        match (self, other) {
+            (Self::Fenced(l0), original::CodeBlockKind::Fenced(r0)) => {
+                l0 == &r0.clone().into_string()
+            }
+            _ => false,
+        }
+    }
+}
+
+impl PartialEq<original::HeadingLevel> for HeadingLevel {
+    fn eq(&self, other: &original::HeadingLevel) -> bool {
+        matches!(
+            (self, other),
+            (HeadingLevel::H1, original::HeadingLevel::H1)
+                | (HeadingLevel::H2, original::HeadingLevel::H2)
+                | (HeadingLevel::H3, original::HeadingLevel::H3)
+                | (HeadingLevel::H4, original::HeadingLevel::H4)
+                | (HeadingLevel::H5, original::HeadingLevel::H5)
+                | (HeadingLevel::H6, original::HeadingLevel::H6)
+        )
+    }
+}
+
+impl From<Range> for std::ops::Range<usize> {
+    fn from(r: Range) -> Self {
+        std::ops::Range {
+            start: r.start.try_into().unwrap(),
+            end: r.end.try_into().unwrap(),
```

Comment:
We might want to explain why the `unwrap()` is fine here.

```suggestion
            start: r.start.try_into().expect("Conversion shouldn't ever fail in practice"),
            end: r.end.try_into()..expect("Conversion shouldn't ever fail in practice"),
```